typescript/src/schema/proto.ts

```diff
@@ -282,6 +283,50 @@ function pagingResponseType(
   return '.google.protobuf.FieldDescriptorProto.Type.' + type;
 }
 
+// Support DIREGAPIC google-discovery API pagination response with map field.
+function pagingMapResponseType(
+  messages: MessagesMap,
+  method: MethodDescriptorProto,
+  rest?: boolean
+) {
+  const pagingfield = pagingField(messages, method, undefined, rest);
+  const outputType = messages[method.outputType!];
+  if (!pagingfield || !pagingfield.type || !rest || !outputType.nestedType) {
```

Comment:
`!pagingfield || !pagingfield.type` can probably be replaced with `!pagingfield?.type`  (the `?.` operator will return `undefined` if `pagingfield` is `undefined`)

Reply:
👍 thanks for the explanation.